tmol/kinematics/move_map.py

```diff
@@ -0,0 +1,466 @@
+        the pose_selection tensor only; if both are not None, then the settings tensor will be indexed by
+        both the pose_selection tensor and the root_jump_selection tensor.
+        """
+        self._assert_valid_pose_block_dof_selection(
+            pose_selection, root_jump_selection, dof_selection, n_movable_jump_dof_types
+        )
+        self._set_val_and_mask_for_3dim(
+            pose_selection, root_jump_selection, dof_selection, "root_jump_dof", value
+        )
+
+    def set_move_atom_dof_for_blocks(
+        self,
+        pose_selection: Union[Tensor, int],
+        block_selection: Optional[Union[Tensor, int]] = None,
+        atom_selection: Optional[Union[Tensor, int]] = None,
+        dof_selection: Optional[Union[Tensor, int]] = None,
+        value: bool = True,
+    ):
+        """Enable or disable partiular atom dofs for a particular set of blocks on particular poses.
+
+        Either only "pose_selection" should be not None or all four "selection" variables should be
+        not None; in the former case, the settings tensor, self.move_atom_dof, will be indexed
+        solely by the pose_selection tensor; in the latter case, the settings tensor will be indexed
+        by all four selection tensors.
+
+        This function offers the finest grain control over which dofs should be minimized and
+        settings made using this function will override any settings made using the other
+        settings tensors.
+        """
+        if block_selection is None:
+            assert atom_selection is None
+            assert dof_selection is None
+            self.move_atom_dof[pose_selection] = value
+            self.move_atom_dof_mask[pose_selection] = True
+        else:
+            self.move_atom_dof[
+                pose_selection, block_selection, atom_selection, dof_selection
+            ] = value
+            self.move_atom_dof_mask[
+                pose_selection, block_selection, atom_selection, dof_selection
+            ] = True
+
+    @property
+    def n_poses(self):
+        return self.move_jumps.shape[0]
+
+    @property
+    def max_n_jumps(self):
+        return self.move_jumps.shape[1]
+
+    @property
+    def max_n_blocks(self):
+        return self.move_mcs.shape[1]
+
+    @property
+    def max_n_named_torsions(self):
+        return self.move_named_torsion.shape[2]
+
+    @property
+    def max_n_atoms_per_block(self):
+        return self.move_atom_dof.shape[2]
+
+    def __init__(
+        self,
+        n_poses: int,
+        max_n_blocks: int,
+        max_n_named_torsions: int,
+        max_n_atoms_per_block: int,
+        device: Optional[torch.device] = None,
+    ):
+        if device is None:
+            device = torch.device("cpu")
+
+        self.move_all_jumps = False
+        self.move_all_root_jumps = False
+        self.move_all_mc = False
+        self.move_all_sc = False
```

Comment:
how does this handle torsion only, versus torsion + bond + angle?  (what is default)